### DISCUSSION TITLE
Deserialization of a JSON object

### DISCUSSION THREAD

**Discussion by ahmed711** [cue-lang/cue#1861] [Q&A] [answered]
Hi all, 
I am new to Cue and I was looking for the right way to deserialize a json object that is passed as a tag from the command line
what I found till now is the following 
```
import "encoding/json" 

#data: #@tag(passedString)#
#file: {
 json.Unmarshal(#data) 
}
```
then from the command line: 
`cue vet -t passedString="Serialized JSON" validation_file file_to_be_validated`

when I do that I receive the following error: 
`no tag for "passedString"`

**Comment by slewiskelly** [comment, marked as answer]
Is this what you need?

```cue
package foo

import "encoding/json"

_data: _ @tag(passedString)

#file: json.Unmarshal(_data)
```

```json
{
    "foo": "bar"
}
```

```console
$ cue vet -d '#file' -t passedString='{"foo": "bar"}' foo.cue foo.json
$ cue vet -d '#file' -t passedString='{"foo": "baz"}' foo.cue foo.json
foo: conflicting values "baz" and "bar":
    ./foo.json:2:12
    json:1:9
```

**Reply by ahmed711** [reply]
Thanks 
That should do it
